src/js/Tabs.js

```diff
@@ -1,24 +1,63 @@
 /*global module, require*/
 const oDom = require('o-dom');
 
-function Tabs(rootEl, config) {
+class Tabs {
 
-	const tabsObj = this;
-	let tabEls;
-	let tabpanelEls;
-	let updateUrl = (rootEl.getAttribute('data-o-tabs-update-url') !== null);
-	let selectedTabIndex = -1;
+	constructor(rootEl, config) {
+		this.rootEl = rootEl;
+		this.rootEl.setAttribute('data-o-tabs--js', '');
 
-	function getTabTargetId(tabEl) {
+		this.updateUrl = (rootEl.getAttribute('data-o-tabs-update-url') !== null);
+		this.selectedTabIndex = -1;
+
+
+		this.tabEls = this.rootEl.querySelectorAll('[role=tab]');
+		this.tabpanelEls = this.getTabPanelEls(this.tabEls);
+
+		this.boundClickHandler = this.clickHandler.bind(this);
+		this.rootEl.addEventListener('click', this.boundClickHandler, false);
+		this.boundKeyPressHandler = this.keyPressHandler.bind(this);
+		this.rootEl.addEventListener('keypress', this.boundKeyPressHandler, false);
+		this.boundHashChangeHandler = this.hashChangeHandler.bind(this);
+		window.addEventListener('hashchange', this.boundHashChangeHandler, false);
+
+		if (!config) {
+			config = {};
+			Array.prototype.forEach.call(this.rootEl.attributes, function(attr) {
+				if (attr.name.indexOf('data-o-tabs') === 0) {
```

Comment:
This might be simpler to understand using the `String.prototype.includes` method: `if (attr.name.includes('data-a-tbs'))`.